core/storage/question/gae_models.py

```diff
@@ -185,6 +185,13 @@ class QuestionSkillLinkModel(base_models.BaseModel):
     # The difficulty of the skill.
     skill_difficulty = ndb.FloatProperty(required=True, indexed=True)
 
+    @staticmethod
+    def get_deletion_policy():
+        """Question skill link should be deleted only if all associated skills
+        or questions are deleted.
+        """
+        return base_models.DELETION_POLICY.KEEP_IF_PUBLIC
```

Comment:
This doesn't really match the docstring? Is this model actually user-related?

Reply:
I was unsure about this too, it is connected undirectly if the user owns private skill and question that are linked and he requests deletion this model should also be deleted then, thus this policy.

Reply:
I don't think skills and questions are "owned" -- @aks681, are there private skills? I thought we weren't going to have those.

And for questions, I think we only have public questions or "question suggestions", but no private questions. @aks681 @vinitamurthi @DubeySandeep can you confirm please?

(The reason explorations are different is because they can take quite a while to create. But questions, skills, etc. -- those should really be quick one-off things. They're also communal -- especially skills -- and they should not "belong to" someone. We should probably update the design doc accordingly once we get the above confirmations, if this isn't clear there.)

Reply:
As @vojtechjelinek said, there are unpublished and published skills. Though, once a skill is published, it cannot be unpublished. So, before the skill is published, it can be considered "private".
For questions, yeah, there are no private/published questions. All created questions can be accessed by learners.

Reply:
We do have "question suggestions" which is not private as it's accessible to reviewers as well as "maybe" in future contributors will be able to see existing suggested questions so that they don't suggest duplicate questions.

Reply:
OK -- I am going to suggest that this is "do not delete" (at least, not in response to user deletion). I'll work with @aks681 to make skills not exist in "private" form since, thinking about it more, that doesn't really make sense. Could you please make sure to update the design doc accordingly, too?

Please make sure that the docstring aligns with the implementation, too. I don't think it's really clear at the moment since it doesn't really explain anything related to "keep if public".

Reply:
Done